go/cmd/dolt/commands/rebase.go

```diff
@@ -0,0 +1,369 @@
+	if strings.Contains(message, "interactive rebase completed") {
+		cli.Println("Successfully rebased and updated refs/heads/" + branchName + ".")
+	} else if strings.Contains(message, "interactive rebase aborted") {
+		cli.Println("Interactive rebase aborted.")
+	} else {
+		rebasePlan, err := getRebasePlan(cliCtx, sqlCtx, queryist, apr.Arg(0), branchName)
+		if err != nil {
+			return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+		}
+
+		if rebasePlan == nil || rebasePlan.Steps == nil {
+			rows, err := GetRowsForSql(queryist, sqlCtx, "CALL DOLT_REBASE('--abort');")
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			status, err := getInt64ColAsInt64(rows[0][0])
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			if status == 1 {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(errors.New("error: "+rows[0][1].(string))), usage)
+			}
+
+			cli.Println("Interactive rebase aborted.")
+		} else {
+			err = insertRebasePlanIntoDoltRebaseTable(rebasePlan, sqlCtx, queryist)
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+
+			rows, err := GetRowsForSql(queryist, sqlCtx, "CALL DOLT_REBASE('--continue');")
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			status, err := getInt64ColAsInt64(rows[0][0])
+			if err != nil {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(err), usage)
+			}
+			if status == 1 {
+				return HandleVErrAndExitCode(errhand.VerboseErrorFromError(errors.New("error: "+rows[0][1].(string))), usage)
+			}
+
+			cli.Println("Successfully rebased and updated refs/heads/" + branchName + ".")
+		}
+	}
+
+	return HandleVErrAndExitCode(nil, usage)
+}
+
+// constructInterpolatedDoltRebaseQuery generates the sql query necessary to call the DOLT_REBASE() function.
+// Also interpolates this query to prevent sql injection.
+func constructInterpolatedDoltRebaseQuery(apr *argparser.ArgParseResults, rebaseBranchName string) (string, error) {
+	var params []interface{}
+	var args []string
+	var checkoutQuery string
+
+	if apr.NArg() == 1 {
+		params = append(params, apr.Arg(0))
+		args = append(args, "?")
+	}
+	if apr.Contains(cli.InteractiveFlag) {
+		args = append(args, "'--interactive'")
+	}
+	if apr.Contains(cli.ContinueFlag) {
+		args = append(args, "'--continue'")
+	}
+	if apr.Contains(cli.AbortParam) {
+		args = append(args, "'--abort'")
+	}
+
+	/*if apr.ContainsAny(cli.ContinueFlag, cli.AbortParam) {
+		checkoutQuery = fmt.Sprintf("CALL DOLT_CHECKOUT('dolt_rebase_%s'); ", rebaseBranchName)
+	}*/
+
+	query := fmt.Sprintf("%sCALL DOLT_REBASE(%s);", checkoutQuery, strings.Join(args, ", "))
+	interpolatedQuery, err := dbr.InterpolateForDialect(query, params, dialect.MySQL)
+	if err != nil {
+		return "", err
+	}
+
+	return interpolatedQuery, nil
```

Comment:
(minor) you could simplify this code a little bit and get rid of those last five lines in the function if you just directly returned the results from `dbr.InterpolateForDialect`. It doesn't change the behavior materially, but it's nice to save five lines of code in this file!